### DISCUSSION TITLE
One execution per queue/job class

### DISCUSSION THREAD

**Discussion by airled** [sidekiq/sidekiq#6269] [Q&A] [answered]
Hi. I've tried to search for this in different places but I can't find exactly what I need. So I have several jobs, say `Job1` and `Job2`. And I have a sidekiq process with concurrency set to 2. I need sidekiq to handle simultaneously one job per job class. So simultaniously `{job1, job2}` is ok. But running `{job1, job1}` or `{job2, job2}` at the same time is not ok. Yes, I understand that I can use `queue1` for `Job1` and `queue2` for `Job2` and then run two sidekiq processes using concurrency 1 with `-q queue1` for first process and `-q queue2` for second. But what if I have such jobs, say, 20? I don't want to run 20 sidekiq processes for each job. Is it possible somehow to handle it within a single sidekiq process? Thanks in advance.

**Comment by mperham** [comment]
A single threaded Capsule is one possibility but I’d advise you to author your jobs in a more concurrent-friendly fashion.

**Comment by freemanoid** [comment, marked as answer]
This sidekiq extension implements exactly what you described https://github.com/ixti/sidekiq-throttled

**Comment by airled** [comment]
@mperham @freemanoid Thank you so much for your responses. Seems like both Capsules and Throttle extension are exactly what I need. @mperham I set freemanoid's post as answer since the extension is more flexible. But using Capsules suits better for my goal.
